internal/datastore/common/sql.go

```diff
@@ -580,19 +702,73 @@ func (tqs QueryExecutor) ExecuteQuery(
 		limit = *queryOpts.Limit
 	}
 
-	toExecute := query.limit(limit)
+	if limit < math.MaxInt64 {
+		query = query.limit(limit)
+	}
+
+	toExecute := query
+
+	// Set the column names to select.
+	columnNamesToSelect := make([]string, 0, 8+len(query.extraFields))
+
+	columnNamesToSelect = checkColumn(columnNamesToSelect, query.schema.ColumnOptimization, query.filteringColumnTracker, query.schema.ColNamespace)
```

Comment:
this logic is essentially replicating the logic in QueryRelationships - there's no way to unify those two things?

Reply:
ditto for spanner

Reply:
Not easily; it is because of Spanner that its hard to do: Spanner needs its own executor code (it can't use the one relationships.go because it has different handling of types for caveats and expiration), but by putting the column selection here, we *do* share at least that portion

Reply:
Can column selection be factored out and called from both places? this seems easy to get out of sync.